src/mpyl/steps/deploy/cloudfront_kubernetes_deploy.py

```diff
@@ -0,0 +1,66 @@
+""" Step that uploads static files to s3 and then deploys the docker image produced in the build stage to Kubernetes,
+using HELM."""
+from logging import Logger
+from typing import Dict
+from .kubernetes import DeployKubernetes
+from .. import Step, Meta
+from ...project import Stage
+from ..models import Input, Output, ArtifactType
+from ...utilities.docker import docker_image_tag, git_tag, docker_copy
+from ...utilities.filesystem import delete_directory
+from ...utilities.s3 import S3Client
+
+TMP_FOLDER = 'tmp'
+STATIC_FOLDER = 'static'
+
+
+class CloudFrontKubernetesDeploy(Step):
+
+    def __init__(self, logger: Logger) -> None:
+        super().__init__(logger, Meta(
+            name='CloudFront Kubernetes Deploy',
+            description='uploads the build output to an s3 bucket',
+            version='0.0.1',
+            stage=Stage.DEPLOY,
+        ), produced_artifact=ArtifactType.DOCKER_IMAGE, required_artifact=ArtifactType.DOCKER_IMAGE)
+
+    def execute(self, step_input: Input) -> Output:
+        bucket_name = self.get_bucket_name(step_input)
+        image_name = docker_image_tag(step_input)
+        container_path = f'{step_input.project.name}/{STATIC_FOLDER}'
+        tag = git_tag(step_input)
+
+        docker_copy(logger=self._logger, container_path=container_path, dst_path=TMP_FOLDER,
+                    image_name=image_name)
+        self.upload_to_s3(run_config=step_input.run_properties.config, bucket_name=bucket_name, src_path=TMP_FOLDER,
+                          bucket_root_path=tag)
+        delete_directory(logger=self._logger, dir_name=TMP_FOLDER)
+
+        return DeployKubernetes(self._logger).execute(step_input)
```

Comment:
Sam mentioned not the use the `after` property when calling the Kubernetes deploy step so I'm calling the execute directly here... not sure if this is the preferred approach, let me know!